src/test/java/org/folio/auth/authtokenmodule/AuthTokenTest.java

```diff
@@ -1016,11 +1039,187 @@ public void testAdminHealth()  {
+    var expiredToken = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    var now = Instant.now().getEpochSecond();
+    expiredToken.setExpiresAt(now - 10);
+    ts.checkTokenNotRevoked(expiredToken).onComplete(context.asyncAssertFailure(e -> {
+      assertThat(e.getMessage(), containsString("expired"));
+      assertThat(e.getMessage(), containsString("revoked"));
+    }));
+  }
+
+  @Test
+  public void testStoreSaveApiToken(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    var apiToken = new ApiToken(tenant);
+    ts.saveToken(apiToken).compose(x ->  ts.checkTokenNotRevoked(apiToken))
+      .onComplete(context.asyncAssertSuccess());
+  }
+
+  @Test
+  public void testStoreApiTokenNotFound(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    // A ApiToken which doesn't exist in storage is treated as revoked.
+    var unsavedToken = new ApiToken(tenant);
+    ts.checkTokenNotRevoked(unsavedToken).onComplete(context.asyncAssertFailure(e -> {
+      assertThat(e.getMessage(), containsString("revoked"));
+    }));
+  }
+
+  @Test
+  public void testApiTokenRevoked(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    var apiToken = new ApiToken(tenant);
+    ts.saveToken(apiToken)
+      .compose(x -> ts.checkTokenNotRevoked(apiToken))
+      .compose(x -> ts.revokeToken(apiToken))
+      .onComplete(context.asyncAssertSuccess())
+      .compose(x -> ts.checkTokenNotRevoked(apiToken))
+      .onComplete(context.asyncAssertFailure(e -> {
+        assertThat(e.getMessage(), containsString("revoked"));
+      }));
+  }
+
+  @Test
+  public void testStoreRefreshTokenSingleUse(TestContext context) {
+    var ts = new RefreshTokenStore(vertx, tenant);
+    // Create and save some tokens.
+    var rt1 = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    var rt2 = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    var rt3 = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    var s1 = ts.saveToken(rt1);
+    var s2 = ts.saveToken(rt2);
+    var s3 = ts.saveToken(rt3);
+
+    Async async = context.async();
+    CompositeFuture.all(s1, s2, s3)
+      .compose(a -> ts.checkTokenNotRevoked(rt2))
+      .onComplete(context.asyncAssertSuccess()) // First check should succeed.
+      .compose(a -> ts.checkTokenNotRevoked(rt2))
+      .onComplete(context.asyncAssertFailure(b -> {
+        assertThat(b.getMessage(), containsString("revoked"));
+      }))
+      .compose(a -> ts.checkTokenNotRevoked(rt1))
+      .onComplete(context.asyncAssertFailure(b -> {
+        assertThat(b.getMessage(), containsString("revoked"));
+      }))
+      .compose(a -> ts.checkTokenNotRevoked(rt3))
+      .onComplete(context.asyncAssertFailure(b -> {
+        assertThat(b.getMessage(), containsString("revoked"));
+        async.complete();
```

Comment:
This async is not needed because context.asyncAssertFailure() has is included.

Reply:
See my comment above. Without `async.complete` I get a timeout. I believe `async.complete` is needed when you have a `CompositeFuture` in the mix,since only those methods which use a composite future have this characteristic. Give the code a try to see what I mean.